### DISCUSSION TITLE
Thank you for the amazing application it literally does everything I ever wanted that I didn't even know was possible.

### DISCUSSION THREAD

**Discussion by yellowbikewithbluehandles** [josephrcox/easy-reddit-downloader#97] [General]
I needed this application because I've been having alot of issues on the healthinsurance subreddit which really helped me fight my health insurance to keep my coverage by constantly reading about different peoples situations and asking my own questions. But a year later I lost coverage again so I was trying to look up all the posts that helped me but it was deleted which sent me into a panic because I lost a lot of info. This made me take screenshots of what I could remember but its impossible to look through and find which led me to look for a solution and I found your comment from 5 months but you had deleted your reddit account so I couldn't send you a message to questions.
My concern is having something like this work forever because I'm scared applications like this will just stop working and I want to just backup and archive things automatically now and the whole subreddit.
I'm also sick of being a leech when people like you are making such cool applications. How did you learn to build something like this, and what would you recommend to learn to be able to create and help maintain these type of things so I can make sure they never go away?
Also from someone completely ignorant on the topic of coding why node.js vs python why would one choose one over the other for these types of applications?
Sorry about the barrage of questions I didn't want to miss a opportunity to talk to someone who created exactly what I wanted.
Thank you for everything!

**Comment by josephrcox** [comment]
Hey there! Thank you so much for this message. I want to be clear that I don't think you're a leech and the appeal of me making this open source was so that it could live on and not be dependent on just myself. The biggest hurdle we have with this isn't having a single dev though, I think it's going to be if Reddit changes their APIs as I describe below. 

> How did you learn to build something like this, and what would you recommend to learn to be able to create and help maintain these type of things so I can make sure they never go away?

This is primarily written in Node JS which is Javascript. Javascript, while it has it's problems, is one of the easier languages to learn I think. 
If I had to start from the get go, I would probably try to learn the basics of web development technologies on YouTube or something. 
This project isn't super advanced but it's also not super simple because it involves talking with external APIs. External APIs (the ones run by Reddit) are messy and the majority of work on this app was actually just learning the APIs and reading them. But that also means that this app is fragile and can break very easily if Reddit changes anything about the structure of the responses. 

> Also from someone completely ignorant on the topic of coding why node.js vs python why would one choose one over the other for these types of applications?

Python honestly is probably a better language to have written this in as it's simpler, requires less boilerplate, has fewer required dependencies, has better typing, and is likely faster. The only reason I wrote this in Node JS was because I recently wrote another app talking to the Reddit APIs, and I wrote that in Javascript as well. This originally started as a fork of that in the early early versions so it was just organic. If I rewrote it from scratch, I probably would write it in Python or Go. 

Thank you so much again! cheers!

**Reply by yellowbikewithbluehandles** [reply]
Thank you so much for the response and advice and giving me a place to start. Please have a good day/night wherever you are from :)!
